File: hw/chavez/u3/hw25TemplateMethod/TemplateMethod/JavaScript/model/Coffee.js
const CaffeineBeverage = require('./CaffeineBeverage');
const readline = require('readline');

class Coffee extends CaffeineBeverage {
  async brew() {
    console.log("Dripping coffee through filter");
  }

  async addCondiments() {
    console.log("Adding sugar and milk");
  }

  async wantsCondiments() {
    const answer = await this.getUserInput("Would you like milk and sugar with your coffee (y/n)? ");
    return answer.trim().toLowerCase().startsWith("y");
  }

  getUserInput(question) {
    const rl = readline.createInterface({
      input: process.stdin,
      output: process.stdout
    });
    return new Promise(resolve => {
      rl.question(question, (ans) => {
        rl.close();
        resolve(ans);
      });
    });
  }
}

module.exports = Coffee;

File: hw/chavez/u3/hw25TemplateMethod/TemplateMethod/Python/model/coffee.py
from model.caffeine_beverage import CaffeineBeverage

class Coffee(CaffeineBeverage):
    def brew(self):
        print("Dripping coffee through filter")

    def add_condiments(self):
        print("Adding sugar and milk")

    def wants_condiments(self):
        answer = input("Would you like milk and sugar with your coffee (y/n)? ").strip().lower()
        return answer.startswith("y")